plugins/org.eclipse.xtend.ide.common/src/org/eclipse/xtend/ide/common/highlighting/XtendHighlightingCalculator.java

```diff
@@ -384,6 +384,7 @@ public void acceptSemanticLineBreak(int charCount, RichStringLiteral origin, boo
 		@Override
 		public void acceptTemplateLineBreak(int charCount, RichStringLiteral origin) {
 			resetCurrentOffset(origin);
+			acceptor.addPosition(currentOffset, charCount, TEMPLATE_LINE_BREAK);
```

Comment:
I don't think it will highlight twice, it's either one or the other case. 
Also 'acceptSemanticLineBreak' looks wrong to me, as it seem to mark semantic NLs as template NLs and POSSIBLE_LINE_BREAK for those that where controlStructureSeen is true (that method also doesn't look very accurate).